modules/stochastic_tools/doc/content/modules/stochastic_tools/examples/gaussian_process_surrogate.md

```diff
@@ -0,0 +1,105 @@
+
+
+!media gaussian_process_uniform_1D_tuned.svg id=1D_tuned caption=Surrogate model for 1D problem $\bar{T}(q)$ using tuned hyperparameters.
+
+Inspection of the final hyperparameter values after tuning gives $\lbrace \sigma_n=3.79E-6, \sigma_f=3.87 , \ell_q=4.59 \rbrace$, the significant increase in the length scale $\ell_q$ is a primary factor in the improved fit.
+
+
+## 2D Problem
+
+Next the idea is extended to two input dimensions and attempt to model the QoI behavior due to varying $k$ and $q$, while fixing values of $\lbrace L=0.03, T_{\infty}=300 \rbrace$.
+
+Due to the increased dimensionality, 50 training samples were selected from $q \in [9000,11000]$ and $k \in [1,10]$. An extra hyperparameter $\ell_k$ is also added to the set of hyperparameters to be tuned.  
+
+!listing examples/surrogates/gaussian_process/gaussian_process_uniform_2D_tuned.i block=Samplers Trainers Covariance
+
+As was done in the 1D case above, first the surrogate was fitted using fixed hyperparameters $\lbrace \sigma_n=1E-3, \sigma_f=10, \ell_q=0.38971 , \ell_k=0.38971 \rbrace$ (hyperparameter tuning disabled by setting `tune=false` option). The QoI surface is shown in [2D_untuned], with the color map corresponding to the surrogate model's uncertainty at that point. As was the case in the 1D mode, uncertainty is highest at points furthest from training data points, and the overall response deviates strongly from the expected $\frac{q}{k}$ behavior predicted.  
+
+!media gaussian_process_uniform_2D.svg id=2D_untuned caption=Surrogate model for 2D problem $\bar{T}(q,k)$ using fixed hyperparameters.
+
+Hyperparameter tuning is then enabled by setting `tune=true`, and the surrogate recreated with trained hyperparameters. [2D_tuned] shows the fit with automatically tuned hyperparameters using the same data set. This results in a fit that much better captures the $\frac{q}{k}$ nature of the QoI response, with "high" uncertainty occurring primarily in extrapolation zones.
+
+!media gaussian_process_uniform_2D_tuned.svg id=2D_tuned caption=Surrogate model for 2D problem $\bar{T}(q,k)$ using tuned hyperparameters.
+
+## Full 4D Problem
+
+This idea is then extended naturally to the full problem in which the parameter set $\lbrace k, q, L, T_{\infty} \rbrace$ is modeled. Training occurs using $500$ training points. Evaluation occurs by sampling the surrogate $10000$ times with perturbed inputs, with results shown in histogram form in [hist]. While the $10000$ evaluation points are sampled from the normal distribution in [param_table],the $500$ training data points are sampled from the uniform distributions. Sampling the training data from the normal distribution can result in an imbalance of data points near the mean, causing poor performance in outlying regions.
```

Comment:
It would be good to mention what input file is running this example. You don't need to show a `!listing` if you don't want.